### PR TITLE
Minor simplification of the type ref walking code

### PR DESCRIPTION
**Pull request by vitek-karas**
Make the walker a true use-once class.

### CODE REVIEW

**Comment by vitek-karas** [comment]
I was reading the PR once more ... and I just had to do this :wink:

File: src/linker/Linker/TypeReferenceWalker.cs [hunk 1]
```diff
@@ -16,21 +16,20 @@ abstract class TypeReferenceWalker
 	{
 		protected readonly AssemblyDefinition assembly;
 
-		HashSet<TypeReference>? visited;
-		protected HashSet<TypeReference> Visited => visited ?? throw new InvalidOperationException ();
+		protected HashSet<TypeReference> Visited { get; init; } = new HashSet<TypeReference> ();
```
Comment (sbomer) [inline src/linker/Linker/TypeReferenceWalker.cs:19]: Why does it need to be init?

**Review by sbomer — APPROVED** [review]
Nice, thank you!

File: src/linker/Linker/TypeReferenceWalker.cs (hunk 1 above)
Comment (vitek-karas) [reply src/linker/Linker/TypeReferenceWalker.cs:19]: It doesn't - it just "reads" right - I'll change it.

**Pull request merged**